design.md

```diff
@@ -0,0 +1,47 @@
+# Design
+
+The idea behind the Health Operator is to have a system that provides a clear view of the health of a system. Probes are a 
+good way to health check containers but they're not good enough to validate the state or how healthy an application is.
+Health checking an application is not just checking that a service responds (200 OK), but it might need a deeper analysis. For
+example, you might want to define that an application is healthy if certain conditions are met, this operator is the foundation
+of how to define and implement those conditions and how to expose them.
+
+Health checks are basically a set of CronJobs that validate certain conditions at certain intervals. This effort is about
+automating the lifecycle of these checks.
+
+There are 3 different use cases:
+
+* Kubernetes service that has to be health checked (`annotation healthcheck=true`)
+* Ad-hoc check that executes one or more than one health check (kind: HealthCheck)
```

Comment:
Exactly, this could definitely be used to run stuff like conformance tests on handmade clusters